packages/web/src/contexts/AuthContext.js

```diff
@@ -77,24 +127,76 @@ const AuthContext = createContext({
 	...INITIAL_STATE,
 	login: () => {},
 	logout: () => {},
+	register: () => {},
+	validateEmail: () => {},
+	validateUsername: () => {},
 })
 
 const AuthContextProvider = props => {
 	const { children } = props
 	const [state, dispatch] = useReducer(reducer, { ...INITIAL_STATE })
 
+	const validateEmail = useCallback(async email => {
+		let response = null
+
+		try {
+			response = await fetch(`/api/users/validate-email?email=${email}`)
```

Comment:
Stupid Question:

Where is this api located? I might be missing something but I don't see this in our project. I'm assuming it is a firestore api, but im having trouble finding documentation on it,

Reply:
Nevermind, I see this now. It is in a subfolder of pages

Reply:
Yup, this uses [Next's API routes](https://nextjs.org/docs/api-routes/introduction). They're quite handy in a lot of circumstances, though each endpoint runs within its own Lambda. This means they have to be stateless, which can be a huge pain-in-the-behind when you need to deal with caching and such.